services/datamanager/builtin/builtin.go

```diff
@@ -530,21 +532,41 @@ func getAllFilesToSync(dir string, lastModifiedMillis int) []string {
 		if info.IsDir() {
 			return nil
 		}
-		// If a file was modified within the past lastModifiedMillis, do not sync it (data
-		// may still be being written).
-		timeSinceMod := clock.Since(info.ModTime())
-		// When using a mock clock in tests, this can be negative since the file system will still use the system clock.
-		// Take max(timeSinceMod, 0) to account for this.
-		if timeSinceMod < 0 {
-			timeSinceMod = 0
-		}
-		isStuckInProgressCaptureFile := filepath.Ext(path) == datacapture.InProgressFileExt &&
-			timeSinceMod >= defaultFileLastModifiedMillis*time.Millisecond
-		isNonCaptureFileThatIsNotBeingWrittenTo := filepath.Ext(path) != datacapture.InProgressFileExt &&
-			timeSinceMod >= time.Duration(lastModifiedMillis)*time.Millisecond
-		isCompletedCaptureFile := filepath.Ext(path) == datacapture.FileExt
-		if isCompletedCaptureFile || isStuckInProgressCaptureFile || isNonCaptureFileThatIsNotBeingWrittenTo {
-			filePaths = append(filePaths, path)
+
+		// Use the lsof command to get all the open files in the directory, and
+		// only retrieve the 9th column which is the "NAME" of the file
+		outBytes, err := exec.Command("lsof", "+d", filepath.Dir(info.Name()), "|", "awk", "'{print $9}'").Output()
```

Comment:
I think exec.Command has a very big performance overhead and should generally be avoided, since it's spinning off a whole separate process. Maybe it's fast enough that we don't care, but I think if we go the lsof route I think it's worth testing/timing this on an rpi to confirm.

I think this also suffers from the problem you describe here:

> Decided to use lsof directly instead of a [package](https://github.com/shirou/gopsutil#process-class) since using the open_files method would restrict us to only being able to use open file descriptors in Linux. Open to suggestions.

Since `lsof` is not a command in windows (I think). But I think that's fine and we're ok with only supporting Linux/Mac for now? @agreenb can confirm there

Reply:
Confirming that fine with only supporting on Linux/Mac for now, agreed on the points to check for performance overhead

Reply:
Okay, so I did test this and it is very time consuming. I also tested with [OpenFiles](https://github.com/shirou/gopsutil) and it seems like it's equally time consuming there as well. So much so that it noticeable degrades the performance of the data manager. I'm going to continue searching for a different solution.